### PR TITLE
basic: use _unlocked() stdio in strip_tab_ansi()

### PR DESCRIPTION
**Pull request by vcaputo**
Trivial performance boost by explicitly bypassing the implicit
locking of stdio.

This significantly affects common cases of `journalctl` usage:

 Before:
```
  # time ./journalctl -b -1 > /dev/null
   real    0m26.628s
   user    0m26.495s
   sys     0m0.125s

  # time ./journalctl -b -1 > /dev/null
   real    0m27.069s
   user    0m26.936s
   sys     0m0.134s

  # time ./journalctl -b -1 > /dev/null
   real    0m26.727s
   user    0m26.607s
   sys     0m0.119s
```
 After:
```
  # time ./journalctl -b -1 > /dev/null
   real    0m23.394s
   user    0m23.244s
   sys     0m0.142s

  # time ./journalctl -b -1 > /dev/null
   real    0m23.283s
   user    0m23.160s
   sys     0m0.121s

  # time ./journalctl -b -1 > /dev/null
   real    0m23.274s
   user    0m23.125s
   sys     0m0.144s
```
Fixes https://github.com/systemd/systemd/issues/6341

### CODE REVIEW

File: src/basic/string-util.c [hunk 1]
```diff
@@ -645,21 +645,21 @@ char *strip_tab_ansi(char **ibuf, size_t *_isz) {
                         else if (*i == '\x1B')
                                 state = STATE_ESCAPE;
                         else if (*i == '\t')
-                                fputs("        ", f);
+                                fputs_unlocked("        ", f);
```
Comment (poettering) [inline src/basic/string-util.c:653]: looks excellent, but could you add a brief line about this aftet the open_memstring() line? something along the lines of "please note we use the 'unlocked' version of the stdio calls, as we know we are the only ones accessing the STDIO objects as we just allocated them here, and this does make quite a performance difference"... or so?

Comment (vcaputo) [reply src/basic/string-util.c:653]: Comment added

**Comment by poettering** [comment]
excellent.

**Comment by crrodriguez** [comment]
glibc head does a similar optimization globally by default anyways. _unlocked variants are used if the program is running single threaded.
https://sourceware.org/git/?p=glibc.git;a=commit;h=d2e04918833d90bae7fe5410bb70a045bbd2b64d

**Comment by poettering** [comment]
@crrodriguez while that is a great optimization in glibc this PR actually also optimizes things for the threaded case: if we know that no other thread could possible have had access to a FILE* object we can avoid the locking even in multi-threaded environments.

And note that most non-trivial programs end up bein multithreaded involuntarily since libraries and such might end up forking something off the background...

**Pull request merged**